_posts/2021-09-30-network-dsl.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: post
+title: "A DSL for Network Specification"
+categories: projects
+tags: projects
+---
+
+We aim to design a high-level, declarative specification of network connectivity
+for use in simulations of morphologically detailled neurons. In particular, such
+a description should take the form of a domain-specific language (DSL) and an
+implementation in Arbor [1]. The project includes a study of the design space
+and prior art [eg 2, 3], as well as the use of connectivity rules in existing
+simulators and the requirements of neuro-scientists. From this a model should be
+developed that captures practical users' needs while building upon a composable
+structure.
+
+1. https://github.com/arbor-sim/arbor
+2. https://www.ncbi.nlm.nih.gov/pubmed/22437992
+3. https://www.neuroml.org/
+
+We are estimating this to be a Master's or Bachelor's project, but depending on
```

Comment:
I think someone with a bit of neuroscientific modelling experience would be a huge plus, nothing beats experience and some knowledge of actual needs. I'd bump the 'requirement' therefore to MSc/PhD, a side project in the latter case.

Reply:
You are probably closer to the mark. I tend to consistently underestimate the complexity of these projects.

Reply:
An additional remark: This can be made _much_ simpler by leaning more into NML style network description and requesting less original research.